File: C#/Strategy/Quiz/Notification.cs
using Strategy.Quiz.Interfaces;

namespace Strategy.Quiz;

public sealed class Notification
{
    private readonly INotificationStrategy _notificationStrategy;

    public Notification(INotificationStrategy notificationStrategy)
        => _notificationStrategy = notificationStrategy;

    public void SendNotification(string message)
        => _notificationStrategy.SendNotification(message);
}

File: PHP/src/Behavioral/Strategy/ExerciseOne/Notification.php
<?php

namespace DesignPatterns\Behavioral\Strategy\ExerciseOne;

use DesignPatterns\Behavioral\Strategy\ExerciseOne\Models\User;
use DesignPatterns\Behavioral\Strategy\ExerciseOne\Interfaces\NotificationInterface;

class Notification 
{
    /**
     * @var NotificationInterface
     */
    private NotificationInterface $notificationStrategy;

    /**
     * @param NotificationInterface $notificationStrategy
     */
    public function __construct(NotificationInterface $notificationStrategy)
    {
        $this->notificationStrategy = $notificationStrategy;
    }

    /**
     * @param User $user
     * @param String $message
     * @return void
     */
    public function sendNotification(User $user, String $message): void
    {
        $this->notificationStrategy->sendNotification($user, $message);
    }
}